dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIExtension.java

```diff
@@ -170,4 +173,31 @@ public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscover
         AnnotatedTypeWrapper<T> wrapper = new AnnotatedTypeWrapper<T>(beanManager, processAnnotatedType.getAnnotatedType(), interceptedClass, interceptedMethods);
         processAnnotatedType.setAnnotatedType(wrapper);
     }
+
+    // Move the FaultToleranceInterceptor up before the Transactional interceptors if required
+    public void afterTypeDiscovery(@Observes AfterTypeDiscovery atd) {
```

Comment:
This would potentially observe every single class. I think you need to add `@WithAnnotations({FaultToleranceAnnotation.class})`. Note that `FaultToleranceAnnotation` is the artificial annotation we add when we have already decided that the FaultToleranceInterceptor is needed. If observing that doesn't work then observing all FT annotations should work, the same as processAnnotatedType...

```
public <T> void processAnnotatedType(@Observes @WithAnnotations({ Asynchronous.class, Fallback.class, Timeout.class, CircuitBreaker.class, Retry.class,
                                                                       Bulkhead.class }) ProcessAnnotatedType<T> processAnnotatedType,
BeanManager beanManager) {
```

Reply:
This is not a valid concern